src/main/java/com/uber/cadence/internal/sync/WorkflowStubImpl.java

```diff
@@ -136,9 +138,21 @@ private StartWorkflowExecutionParameters getStartWorkflowExecutionParameters(
     }
     p.setInput(dataConverter.toData(args));
     p.setWorkflowType(new WorkflowType().setName(workflowType.get()));
+    p.setMemo(convertMemoFromObjectToBytes(o.getMemo()));
     return p;
   }
 
+  private Map<String, byte[]> convertMemoFromObjectToBytes(Map<String, Object> memoFromOption) {
+    if (memoFromOption == null) {
+      return null;
+    }
+    Map<String, byte[]> memo = new HashMap<>();
+    for (Map.Entry<String, Object> item : memoFromOption.entrySet()) {
+      memo.put(item.getKey(), dataConverter.toData(item.getValue()));
```

Comment:
I would catch the exception and add information about which field of the memo failed serialization.

Reply:
updated